crates/autopilot/src/run_loop.rs

```diff
@@ -74,14 +79,81 @@ impl RunLoop {
 
         // Shuffle so that sorting randomly splits ties.
         solutions.shuffle(&mut rand::thread_rng());
-        solutions.sort_unstable_by(|left, right| left.1.score.total_cmp(&right.1.score));
+        solutions.sort_unstable_by_key(|solution| solution.1.score);
 
         // TODO: Keep going with other solutions until some deadline.
         if let Some((index, solution)) = solutions.pop() {
             // The winner has score 0 so all solutions are empty.
-            if solution.score == 0. {
+            if solution.score == 0.into() {
                 return;
             }
+
+            let auction_id = id;
+            let winner = solution.reward.performance_address;
+            let winning_score = solution.score;
+            let reference_score = solutions
+                .last()
+                .map(|(_, response)| response.score)
+                .unwrap_or_default();
+            let mut participants = solutions
+                .iter()
+                .map(|(_, response)| response.reward.participation_address)
+                .collect::<HashSet<_>>();
+            participants.insert(solution.reward.participation_address); // add winner as participant
+            let prices = auction.prices.clone();
+            let block_deadline = self.current_block.borrow().number
+                + self.submission_deadline
+                + self.additional_deadline_for_rewards;
+            // save order executions for all orders in the solution, except for partial
+            // limit orders. order executions for partial limit orders will be
+            // saved after settling the order onchain.
+            let mut order_executions = vec![];
+            for order_id in &solution.orders {
+                let auction_order = auction
+                    .orders
+                    .iter()
+                    .find(|auction_order| &auction_order.metadata.uid == order_id);
+                match auction_order {
+                    Some(auction_order) => {
+                        let executed_fee = match auction_order.metadata.class {
+                            OrderClass::Limit(LimitOrderClass { surplus_fee, .. }) => {
+                                match auction_order.data.partially_fillable {
+                                    // we don't know the surplus fee in advance. will be populated
+                                    // after the transaction containing the order is mined
+                                    true => ExecutedFee::Surplus(None),
+                                    // calculated by the protocol, therefore we can trust the
+                                    // auction
+                                    false => ExecutedFee::Surplus(surplus_fee),
```

Comment:
Not sure if I should take the `surplus_fee` from the auction or reuse the code from partial limit orders for calculating the `surplus_fee` from the mined transaction. Mostly because the encoding is done in the driver we don't control anymore with colocation.

Reply:
Depends on what `executed_fee` is supposed to mean in this case.
If this is just the value to be used for objective value computations trusting the protocol computed value here is correct. If you are interested in the fee that the user actually ended up paying this value needs to be computed when we index the trade execution.
I believe the former is needed here so this code looks fine to me.